### DISCUSSION TITLE
Logging multiple metrics to hparams tab in tensorboard

### DISCUSSION THREAD

**Discussion by adobles96** [Lightning-AI/pytorch-lightning#8138] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi,

I've been trying to log multiple metrics to the hparam tab in tensorboard but haven't been able to do it. This is what I'm doing (mostly following the [documentation](https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html#logging-hyperparameters)):

```python
class MyModule(pl.LightningModule):
    def __init__(self, config: Dict):
        self.save_hyperparameters(config)
        self.metrics = [
            'hp/best_val_loss',
            'hp/best_val_acc',
        ]
        ...
    
    def on_train_start(self):
        self.logger.log_hyperparams(self.hparams, {k: 0 for k in self.metrics})

    def validation_epoch_end(self, val_outputs):
        metrics = calc_aggregated_metrics(val_outputs)
        for metric, value in metrics.items():
            self.log(metric, value)  # here metrics corresponds to the metrics in self.metrics
            

tb_logger = pl_loggers.TensorBoardLogger('logs/', default_hp_metric=False)
M = MyModule(config)
trainer = pl.Trainer(logger=tb_logger)
trainer.fit(M, train_loader, val_loader)
```

When looking at tensorboard, the `'hp/*'` metrics show up fine in the scalars tab, but in nothing shows up in the hparams tab. What's interesting is that if I comment out the `self.log` line in `validation_epoch_end` then I do see hparams in the hparams tab (but obviously not the metrics), so there seems to be a conflict between logging the metrics and seeing the hyperparameters in the hparams tab. Any ideas as to what I'm doing wrong?

**Comment by Borda** [comment]
@adobles96 mind provide more information what is the implementation of your `metrics = calc_aggregated_metrics(val_outputs)` as it shall be a dictionary of string and number? Also, could you tell us if there is any error message or how your TB looks like?
